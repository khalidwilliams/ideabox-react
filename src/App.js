import React from 'react';
import logo from './logo.svg';
import './App.css';
import IdeaCard from './IdeaCard';

function App() {
  return (
    <div className="App">
      <IdeaCard 
        id={1}
        title="Dogs Should Vote"
        description="Give them rights"
        deleteCard={(id) => {}}
      />
    </div>
  );
}

export default App;
