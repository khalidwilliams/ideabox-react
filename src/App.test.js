import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import { shallow } from 'enzyme';

describe('App', () => {
  test('renders application', () => {
    let shallowApp = shallow(<App />);
    expect(shallowApp).toMatchSnapshot();
  });

  it('should have no cards initially', () => {
    let shallowApp = shallow(<App />);
    expect(shallowApp.state('ideas')).toEqual([]);
  });

  it('should remove a card from state when deleteCard is called', () => {
  
  }); 

  it('should add a card to state when addCard is called', () => {
  
  }); 

  it('should modify a card\'s favorite status when toggleFavorite is called', () => {
  
  }); 
});
