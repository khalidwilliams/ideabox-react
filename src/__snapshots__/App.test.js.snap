// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders application 1`] = `
<div
  className="App"
>
  <IdeaCard
    deleteCard={[Function]}
    description="Give them rights"
    id={1}
    title="Dogs Should Vote"
  />
</div>
`;
